src/v/storage/disk_log_impl.cc

```diff
@@ -560,8 +567,9 @@ ss::future<> disk_log_impl::compact(compaction_config cfg) {
     if (config().is_compacted() && !_segs.empty()) {
         f = f.then([this, cfg] { return do_compact(cfg); });
     }
-    return f.then(
-      [this] { _probe.set_compaction_ratio(_compaction_ratio.get()); });
+    return f
+      .then([this] { _probe.set_compaction_ratio(_compaction_ratio.get()); })
+      .finally([guard = std::move(guard)] {});
```

Comment:
co_return? gate guard is for coroutines (this is probably still correct, but we should try to be consistent)

Reply:
I treat this gate guard similar to semaphore units. i.e. it gives you more granular control over that when gate is being left and do not require additional indentations.

Reply:
I've used `try_with_gate` for consistency.